lib/config.module.ts

```diff
@@ -30,24 +30,25 @@ export class ConfigModule {
    * @param options
    */
   static forRoot(options: ConfigModuleOptions = {}): DynamicModule {
+    let config = { ...process.env };
```

Comment:
In this case, Joi will validate all the process variables, even these like `SHELL` or `NODE` and this doesn't make too much sense. What if someone set `allowUnknown` to `false`? In this case, the validation will **always** throw an error. 

Generally, the `validationSchema` option was intended to be used with `dotenv` and `.env` files. Maybe, if we want to allow validating process environment variables too, we should add a flag for this (another option that would force set `allowUnknown` to `false` as well and assign `process.env` to `config` as above).

Reply:
From [nestjs configuration document](https://docs.nestjs.com/techniques/configuration),

> Alternatively, we can use the required() validation method to require that a value must be defined in the environment (.env file or process environment).

Does document wrong until this commit is merged?

Reply:
@ojy6042 There is nothing wrong in the documentation. 

What Kamil is talking about is allowing "unknown" variables in your `.env` file. If you scroll down the [Schema Validation section](https://docs.nestjs.com/techniques/configuration#schema-validation) of the documentation, you'd see a setting `allowUknown`. If the user sets it to false, it will end up always giving errors (if this commit is merged).